spotbugs/src/main/java/edu/umd/cs/findbugs/classfile/impl/ClassFactory.java

```diff
@@ -107,20 +106,22 @@ static IScannableCodeBase createFilesystemCodeBase(FilesystemCodeBaseLocator cod
 
         File file = new File(fileName);
 
-        if (!file.exists()) {
-            throw new FileNotFoundException("File " + file.getAbsolutePath() + " doesn't exist");
-        } else if (!file.canRead()) {
-            throw new IOException("File " + file.getAbsolutePath() + " not readable");
+        if (!file.exists() || !file.canRead()) {
+            return new EmptyCodeBase(codeBaseLocator);
         } else if (file.isDirectory()) {
             return new DirectoryCodeBase(codeBaseLocator, file);
         } else if (!file.isFile()) {
-            throw new IOException("File " + file.getAbsolutePath() + " is not a normal file");
+            return new EmptyCodeBase(codeBaseLocator);
         } else if (fileName.endsWith(".class")) {
             return new SingleFileCodeBase(codeBaseLocator, fileName);
         } else if (fileName.endsWith(File.separator + "jrt-fs.jar")) {
             return new JrtfsCodeBase(codeBaseLocator, fileName);
         } else {
-            return ZipCodeBaseFactory.makeZipCodeBase(codeBaseLocator, file);
+            try {
+                return ZipCodeBaseFactory.makeZipCodeBase(codeBaseLocator, file);
+            } catch (IOException e) {
+                return new EmptyCodeBase(codeBaseLocator);
```

Comment:
I think we need to log the stack trace of `file`, to help debugging.  
We should pay attention when ignore thrown exception, I think.